jms/deployment/src/main/java/io/quarkus/artemis/jms/deployment/camel/CamelJmsRegistryEnhancerProcessor.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.artemis.jms.deployment.camel;
+
+import org.apache.camel.quarkus.core.CamelCapabilities;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.artemis.jms.runtime.camel.CamelContextEnhancer;
+import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.annotations.BuildStep;
+
+public class CamelJmsRegistryEnhancerProcessor {
+    @SuppressWarnings("unused")
+    @BuildStep
+    AdditionalBeanBuildItem addCamelContextEnhancer(Capabilities capabilities) {
+        if (capabilities.isPresent(CamelCapabilities.CORE)) {
+            return AdditionalBeanBuildItem.unremovableOf(CamelContextEnhancer.class);
```

Comment:
I think `CamelContextEnhancer` is not necessary here and `camel-quarkus` should handle it already. Is it possible to remove this class and run the `camel-jms` tests to see if they are still working?

Reply:
Do you men removing the whole build step (i.e. the package `io.quarkus.artemis.jms.[deployment|runtime].camel`?

Reply:
Removing class `CamelJmsRegistryEnhancerProducer` results in failing integration tests for module `quarkus-integration-test-artemis-camel-jms-with-default-and-named`:

```
...
Caused by: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: <default> of type: javax.jms.ConnectionFactory
...
```

Adding `@Named(...)`-annotations to the synthetic beans in addition to removing class `CamelJmsRegistryEnhancerProducer` also results in failing integration tests for module `quarkus-integration-test-artemis-camel-jms-with-default-and-named`

```
...
Caused by: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: <default> of type: javax.jms.ConnectionFactory
...
```

I have attached [patches (`.zip`)]((https://github.com/quarkiverse/quarkus-artemis/files/10208625/patches.zip)) for both versions if you want to take a look yourself. So as far as I can see, there is no automatism on camel's end, hence we have to add the beans to the camel context manually.

Reply:
`<default>` should be the default bean of `ConnectionFactory` ? I'm not sure why you need `from(jms("queue:in").connectionFactory("<default>"))`, just to try `from(jms("queue:in"))` to see if it is working?

Reply:
As far I as know, it's no need to use `connectionFactory("<default>"))` if there is a defautl ben of `ConnectionFactory`

Reply:
Will try, but I think that the behaviour with not providing a connection factory only works if exactly one connection factory is found.

Reply:
@zhfeng removing `.connectionFactory("<default>")` ([Patch (`.zip`)](https://github.com/quarkiverse/quarkus-artemis/files/10209889/removeConnectionFactory.zip)) leads to `quarkus-integration-test-artemis-camel-jms-with-default-and-named` failing:

```
...
Caused by: java.lang.IllegalArgumentException: connectionFactory must be specified
...

```

Reply:
@zhfeng can this discussion be resolved?

Reply:
@zhfeng can this discussion be resolved?

Reply:
@turing85 It's weird and I will try to figure it out. But I'm in holiday these weeks, can you hold this PR and wait for my checking?

Thanks!

Reply:
@zhfeng of course, no problem. Will hold the PR.

Reply:
@zhfeng [as discussed](https://github.com/quarkiverse/quarkus-artemis/pull/125#discussion_r1060581213), I added the `name(...)` s and the toggle for the `CamelContextEnhancer`. Do you thin this PR is ready to be merged?